cub/test/CMakeLists.txt

```
@@ -26,6 +26,9 @@ option(METAL_BUILD_EXAMPLES OFF)
 option(METAL_BUILD_TESTS OFF)
 CPMAddPackage("gh:brunocodutra/metal@2.1.4")
 
+CPMAddPackage(NAME NVTX GITHUB_REPOSITORY NVIDIA/NVTX GIT_TAG release-v3 DOWNLOAD_ONLY True SYSTEM)
+add_subdirectory(${NVTX_SOURCE_DIR}/c ${NVTX_BINARY_DIR}/c)
```

It looks like NVTX does provide CMake targets, it's just a little...odd: https://github.com/NVIDIA/NVTX/tree/release-v3?tab=readme-ov-file#imported-cmake-targets

We'll want the imported targets, so this should be something like:

```suggestion
CPMAddPackage(
  NAME NVTX
  GITHUB_REPOSITORY NVIDIA/NVTX
  GIT_TAG release-v3
  DOWNLOAD_ONLY
  SYSTEM
)
include("${NVTX_SOURCE_DIR}/c/nvtxImportedTargets.cmake")
```

This defines a `nvtx3-cpp` target that we can use from this subdirectory.

Note that `DOWNLOAD_ONLY` doesn't take an arg, it's always off unless the flag is passed to `CPMAddPackage`.

---

Awesome, I knew you would know a better way! Thanks a lot!

---

I've filed NVIDIA/NVTX#94. If that gets implemented we can can replace all of this with the standard

```
CPMAddPackage("gh:NVIDIA/NVTX@release-v3")
```